### DISCUSSION TITLE
Offline support?

### DISCUSSION THREAD

**Discussion by davoutuk** [serverpod/serverpod#862] [General]
I'm a newbie to ServerPod, and was wondering how/what it offers in terms of offline support?  i.e.  a user wants to add/edit data when there is no internet connection.

**Comment by FBanitz** [comment]
We are currently building an app using Serverpod and the same question was brought up to the table. 
Historically we were using Hive for local storage in parallel to server storage. In my knowledge it is not possible to use Hive with Serverpod as the models are generated from the corresponding .yaml file, and, by definition it is not possible to add the needed Hive fields annotations. 
Am I mistaken in my reflection ? Do you know if this is possible with the current version of Serverpod, or if there are any other options for saving data locally ? 
And if not, is there any plan to add this feature in a future release? 
Thanks!

**Reply by FBanitz** [reply]
My bad, it seems like you just can use [custom serializable objects](https://docs.serverpod.dev/concepts/serialization#custom-serializable-objects) to use Hive or another local storage package. Looks like haven't read the doc closely enough 😅.

**Comment by davoutuk** [comment]
Having client side caching of server call data results might enable  the idea of adding/updating data when offline.   For me this is the 'killer' feature that would give the platform market advantage
